clientconn.go

```diff
@@ -1515,6 +1517,20 @@ func (ac *addrConn) tearDown(err error) {
 	if channelz.IsOn() {
 		channelz.RemoveEntry(ac.channelzID)
 	}
+
+	// TODO(deklerk) is this bad?
+	transports := map[int32]transport.ClientTransport{}
```

Comment:
Naive way to get around needing the transports outside the lock (GracefulClose takes the lock), but not wanting to cause data races.

Is there a better way to do this?